### ISSUE TITLE
Add multi-hour production preset for training example

### ISSUE THREAD

**Issue by evgri243** [JetBrains-Research/opaque#51]
## Summary
Add a production preset for `train_causal_lm.py` that runs a multi-hour training with a 4-8B parameter model on real data with multiple noise parameters.

## Configuration
| Parameter | Value | Rationale |
|-----------|-------|-----------|
| Model | Qwen2.5-7B or Llama-3-8B | Production-scale |
| Dataset | Alpaca, Dolly, or domain-specific | Real instruction tuning |
| Samples | Full dataset (50k+) | Real training |
| Batch size | 16 | Memory-efficient with LoRA |
| Epochs | 10-80 | Until convergence or budget exhaustion |
| LoRA rank | 16 | Standard for 7-8B models |
| Noise multipliers | [0.5, 1.0, 2.0, 5.0] | Sweep for utility/privacy tradeoff |
| Clip norm | 1.0 | Standard, or adaptive |
| Target epsilon | 3.0, 8.0 | Practical privacy budgets |

## Multi-run Support
The production preset should support sweeping over noise multipliers:
```bash
# Single run
python train_causal_lm.py --preset production_7b --noise_multiplier 1.0

# Sweep (future: use W&B sweep or simple loop)
for nm in 0.5 1.0 2.0 5.0; do
  python train_causal_lm.py --preset production_7b --noise_multiplier $nm
done
```

## Expected Output
- Training logs with loss, gradient norms, clip rate per step
- Final privacy report (epsilon at target delta)
- W&B dashboard (if `--wandb` enabled)
- Saved model checkpoint

## Acceptance Criteria
- [ ] Preset configures realistic 7-8B training
- [ ] Runs for multiple hours without OOM
- [ ] Privacy budget tracked and reported
- [ ] Results reproducible with fixed seed
- [ ] Documentation: expected runtime, GPU requirements

## Depends on
- #47 (accounting must be in train_causal_lm.py)
- #49 (W&B nice to have)

## Parent
- #65
